src/main/java/impl/AllStrategyImpl.java

```diff
@@ -0,0 +1,27 @@
+package impl;
+
+import strategies.SearchStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class AllStrategyImpl implements SearchStrategy {
+
+    @Override
+    public List<String> search(ArrayList<String> list, Map<String, ArrayList<Integer>> searchMap, String query) {
+        String[] words = query.trim().split("\s");
```

Comment:
The regex expression is repeated in 3 other places. Please think about constant.

Reply:
replaced to interface